### DISCUSSION TITLE
Live Transcription Viewed From Separate Page

### DISCUSSION THREAD

**Discussion by nccscott** [deepgram/community#754] [News + Announcements]
I followed the instructions on this page to set up a live transcription using deepgram which displays the transcription in the browser.  Can someone walk me through how to get that transcription to be displayed on a separate page that is not actively transcribing so that someone else can view it from their smartphone?  https://deepgram.com/learn/live-transcription-mic-browser#handling-the-deepgram-response

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment]
Hey @nccscott, if the person on the smartphone needs to see the transcript in realtime, you can create a new websocket or webrtc session to re-stream the data. If they only need to see it after-the-fact, the simplest solution is to upload the finished transcript to a cloud storage service such as AWS S3 -- the smartphone app can then download the transcript from permanent storage to view it.

**Comment by nccscott** [comment]
Thanks jjmaldonis for responding so quickly!  Yes, it would be to see live.  We are trying to have a live transcription for people who are hard of hearing.  I am not knowledgable at all in all this, is there any direction/advice/example of what it would look like for me to "Create a new websocket" or "webrtc"?

**Comment by jjmaldonis** [comment]
This is a great article to read for an overview of the differences between websockets and WebRTC:  https://medium.com/agora-io/how-does-webrtc-work-996748603141

No matter which option you choose, the first question you need to answer is:
How will the smartphone know which request to listen to? If you can provide this information to the smartphone app, then you can also provide a websocket URL or a WebRTC session for the smartphone app to connect to.

After that, everything is technical details.

I personally would recommend using WebRTC. The language around the technology is technical and confusing, but your use case is simple enough that you can ignore most of the complexity and find a simple example in your programming language of choice to get up and running.

WebRTC also makes it easy for many users on the smartphone app to connect to the same session (e.g. for a live broadcast), whereas a websocket implementation typically requires a different websocket for each client (one of the main use cases for WebRTC is to overcome this one-to-one communication limitation of websockets).

Your setup is very simple: You have one client that is broadcasting messages over the WebRTC session to all other connected clients. The other clients do not need to respond or send messages over the WebRTC session, they only listen for messages. This simple broadcasting mechanism means you don't need to worry about complexities around "who is sending which message" and how to process messages from different clients. (In contrast, consider a chatbot as the most simple next step up in complexity, where two clients need to both send and receive messages.)

WebRTC is typically used for audio and video streams, so most of the information you read will be about those, but it works perfectly well for any type of data, including text data (encoded as UTF-8 binary data).
